eras/conway/impl/src/Cardano/Ledger/Conway/Rules/NewEpoch.hs

```diff
@@ -143,6 +132,24 @@ instance
           def
     ]
 
+  assertions =
+    [ let checkRewardsBalanced es (RewardUpdate dt dr rs_ df _) =
+            Val.isZero $
+              dt <> dr <> toDeltaCoin (sumRewards (es ^. prevPParamsEpochStateL . ppProtocolVersionL) rs_) <> df
+       in PostCondition
+            "Reward updates balance out"
+            ( \(TRC (_, NewEpochState eL _bPrev _bCur es ru _pd _avvm, eNo)) _newState ->
+                if eNo /= succ eL
+                  then True
+                  else case ru of
+                    SNothing -> True
+                    SJust p@(Pulsing _ _) ->
+                      let (ru', _event) = runIdentity $ runReaderT (completeRupd p) testGlobals
```

Comment:
This is most definitely is incorrect!

How would you expect any sort of values that are used for testing be applicable in validation that is used when replaying the chain for mainnet?

Reply:
https://github.com/IntersectMBO/cardano-ledger/issues/5036 is opened in order to track this.